src/IB/IBHydrodynamicForceEvaluator.cpp

```diff
@@ -104,6 +105,51 @@ IBHydrodynamicForceEvaluator::registerStructure(int strct_id,
     bool from_restart = RestartManager::getManager()->isFromRestart();
     if (!from_restart)
     {
+        // Ensure that box is aligned to grid cell sides at coarsest level
+	const int coarsest_ln = 0;
+	Pointer<PatchLevel<NDIM> > coarsest_level = patch_hierarchy->getPatchLevel(coarsest_ln);
+	const Pointer<CartesianGridGeometry<NDIM> > coarsest_grid_geom = coarsest_level->getGridGeometry();
+	const double* const dx_coarsest = coarsest_grid_geom->getDx();
+	const double* const grid_X_lower = coarsest_grid_geom->getXLower();
+	bool modified_box = false;
+	const Eigen::Vector3d box_X_lower_old = box_X_lower;
+	const Eigen::Vector3d box_X_upper_old = box_X_upper;
+	
+	for (int d = 0; d < NDIM; ++d)
+	{
+	    double num_cells_lower = (box_X_lower[d]-grid_X_lower[d])/dx_coarsest[d];
+	    double num_cells_upper = (box_X_upper[d]-grid_X_lower[d])/dx_coarsest[d];
+	    
+	    if (num_cells_lower != (int) num_cells_lower)
```

Comment:
@nishantnangia : This looks suspicious. How about MathUtilities<double>::equalEps(num_cells_lower, floor(num_cells_lower)) ?